crates/server/src/authority/metrics.rs

```diff
@@ -5,16 +5,55 @@
 // https://opensource.org/licenses/MIT>, at your option. This file may not be
 // copied, modified, or distributed except according to those terms.
 
+use std::collections::HashMap;
+
+use crate::authority::{AuthLookup, Authority, LookupControlFlow, ZoneType};
+
 use hickory_proto::op::LowerQuery;
 use hickory_proto::rr::{DNSClass, Record, RecordType};
 use metrics::{Counter, Unit, counter, describe_counter};
 
 pub(super) struct CatalogMetrics {
+    zone_store_metrics: HashMap<&'static str, ZoneLookupMetrics>,
```

Comment:
This should probably be keyed by both the authority's `metrics_label()` string and the `ZoneType`, just in case.

Reply:
I see. That is especially useful for `file` where `ZoneType` can be configured.